src/main/java/com/redhat/devtools/intellij/quarkus/QuarkusDeploymentSupport.java

```diff
@@ -0,0 +1,274 @@
+                        progressIndicator.checkCanceled();
+                        progressIndicator.setText("Adding in ''" + module.getName() + "'' Quarkus deployment dependencies to classpath...");
+                        List<VirtualFile>[] files = toolDelegate.getDeploymentFiles(module, progressIndicator);
+                        LOGGER.info("Adding library to " + module.getName() + " previousHash=" + previousHash + " newHash=" + actualHash);
+                        try {
+                            TelemetryService.instance().action(TelemetryService.MODEL_PREFIX + "addLibrary").send();
+                        } catch (Exception e) {
+
+                        }
+                        addLibrary(model, files);
+                    });
+                    component.setHash(actualHash);
+                    component.setVersion(QuarkusConstants.QUARKUS_DEPLOYMENT_LIBRARY_VERSION);
+                }
+            }
+        }
+        LOGGER.info("ensureQuarkusLibrary ran in " + (System.currentTimeMillis() - start));
+    }
+
+
+    private static boolean isOutOfDated(CompletableFuture<Void> loader) {
+        return loader == null;
+    }
+
+    @NotNull
+    private synchronized static CompletableFuture<Void> internalUpdateClasspathWithQuarkusDeployment(Module module) {
+        CompletableFuture<Void> deploymentSupport = module.getUserData(QUARKUS_DEPLOYMENT_SUPPORT_KEY);
+        if (!isOutOfDated(deploymentSupport)) {
+            return deploymentSupport;
+        }
+        var project = module.getProject();
+        final CompletableFuture<Void> future = new CompletableFuture<>();
+        CompletableFuture.runAsync(() -> {
+            Runnable task = () -> ProgressManager.getInstance().run(new Task.Backgroundable(project, "Adding Quarkus deployment dependencies to classpath...") {
+                @Override
+                public void run(@NotNull ProgressIndicator indicator) {
+                    try {
+                        long start = System.currentTimeMillis();
+                        ProgressIndicator wrappedIndicator = new ProgressIndicatorWrapper(indicator) {
+
+                            @Override
+                            public boolean isCanceled() {
+                                return super.isCanceled() || future.isCancelled();
+                            }
+                        };
+                        updateClasspathWithQuarkusDeployment(module, wrappedIndicator);
+                        long elapsed = System.currentTimeMillis() - start;
+                        LOGGER.info("Ensured QuarkusLibrary in {} ms", elapsed);
+                        future.complete(null);
+                    } catch (CancellationException | ProcessCanceledException e) {
+                        future.cancel(true);
+                    } catch (Exception e) {
+                        LOGGER.error("Error while adding Quarkus deployment dependencies to classpath in '{}'", module.getName(), e);

```

Comment:
actually this won't work, since logging an error should take no other parameters https://www.slf4j.org/api/org/slf4j/Logger.html#error-java.lang.String-java.lang.Throwable-

Reply:
Ok I revert the changes.